File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/application/command/result/RegisterUserResult.java
package dev.starryeye.book_library.application.command.result;

import dev.starryeye.book_library.domain.user.User;

public record RegisterUserResult(
        Long id,
        String username,
        Integer age
) {

    public static RegisterUserResult of(User user) {
        return new RegisterUserResult(
                user.getId(),
                user.getUsername(),
                user.getAge()
        );
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/command/result/RegisterUserResult.kt
package dev.starryeye.book_library.application.user.command.result

import dev.starryeye.book_library.domain.user.User

data class RegisterUserResult(
    val id: Long,
    val username: String,
    val age: Int?,
) {

    companion object {
        fun of(user: User): RegisterUserResult {
            return RegisterUserResult(
                id = user.id!!, // DB 거쳐서 나온 엔티티에는 id 가 항상 존재한다.
                username = user.username,
                age = user.age
            )
        }
    }
}